apps/marketplace/components/SellerSelect/SellerSelect.js

```diff
@@ -0,0 +1,134 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import AUtextInput from '@gov.au/text-inputs/lib/js/react.js'
+import AUbutton from '@gov.au/buttons/lib/js/react.js'
+import findSuppliers from 'marketplace/actions/supplierActions'
+import styles from './SellerSelect.scss'
+
+const SellerSelectView = props => (
+  <div>
+    <label htmlFor={props.id}>Select sellers</label>
+    <AUtextInput
+      id={props.id}
+      placeholder={props.placeholder}
+      onChange={props.handleOnChange}
+      value={props.inputValue}
+      className={props.className}
+    />
+    {props.showSearchButton && <AUbutton onClick={props.handleSearchClick}>Search</AUbutton>}
+  </div>
+)
+
+const SellerSelectResultsView = props => (
+  <ul className={props.className}>
+    {props.sellers.map(seller => (
+      <li key={seller.code}>
+        <a href={`#${seller.code}`} onClick={e => props.handleSellerSelectClick(seller, e)}>
+          {seller.name}
+        </a>
+        {seller.panel && <span className={props.panelBadgeClassName}>✓ Panel</span>}
+        {seller.sme && <span className={props.smeBadgeClassName}>SME</span>}
+      </li>
+    ))}
+  </ul>
+)
+
+let timeoutHandle = null
+
+export class SellerSelect extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      sellers: [],
+      inputValue: ''
+    }
+    this.handleOnChange = this.handleOnChange.bind(this)
+    this.handleSellerSelectClick = this.handleSellerSelectClick.bind(this)
+    this.handleSearchClick = this.handleSearchClick.bind(this)
+  }
+
+  handleOnChange(e) {
+    const keyword = e.target.value
+    this.setState({
+      inputValue: keyword
+    })
+
+    if (timeoutHandle) {
+      clearTimeout(timeoutHandle)
+    }
+
+    timeoutHandle = setTimeout(() => {
```

Comment:
the `setTimeout()` is to prevent flooding the network with requests after the 2 chars minimum. When the `onChange` is fired for the input, it will cancel any unprocessed `setTimeout` currently pending, preventing previous key strokes from impacting on the UI results should network requests not come back in order.